### ISSUE TITLE
prefer-for-of rule false positive when with HTMLCollectionOf<Element>

### ISSUE THREAD

**Issue by plantain-00** [palantir/tslint#2549]
### Bug Report

- __TSLint version__:5.0.0
- __TypeScript version__:2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const children = document.getElementsByClassName("abc");
for (let i = 0; i < children.length; i++) {
    const child = children[i];
    console.log(child);
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:latest",
    "rules": {
        "forin": false,
        "interface-name": [
            false
        ],
        "max-line-length": [
            false
        ],
        "no-var-requires": false,
        "no-console": [
            false
        ],
        "no-string-literal": false,
        "no-reference": false,
        "ordered-imports": [
            false
        ],
        "object-literal-sort-keys": false,
        "variable-name": [
            true,
            "ban-keywords"
        ],
        "no-bitwise": false,
        "member-access": false,
        "arrow-parens": false,
        "array-type": [
            true,
            "array"
        ],
        "max-classes-per-file": [
            false
        ],
        "interface-over-type-literal": false
    }
}
```

#### Actual behavior

```
Expected a 'for-of' loop instead of a 'for' loop with this simple iteration
```

#### Expected behavior

No error, because `HTMLCollectionOf<Element>` can not work with `for...of`:

```ts
const children = document.getElementsByClassName("abc");
for (const child of children) {
    console.log(child);
}
```

```
[ts] Type 'HTMLCollectionOf<Element>' is not an array type or a string type.
```

**Comment by andy-hanson** [comment]
Looks like a dupe of #2021.
